samples/MerQure.Tools.Samples/RetryBusExample/Infra/SampleService.cs

```diff
@@ -0,0 +1,68 @@
+﻿using MerQure.Tools;
+using MerQure.Tools.Configurations;
+using MerQure.Tools.Buses;
+using MerQure.Tools.Samples.RetryBusExample.Domain;
+using MerQure.Tools.Samples.RetryBusExample.Infra;
+using System;
+using System.Collections.Generic;
+
+
+namespace MerQure.Tools.Samples.RetryBusExample.RetryExchangeExample.Infra
+{
+    public class SampleService : ISampleService
+    {
+        public const string BusName = "sample";
+        private readonly IRetryBusService _retryBusService;
+
+        public IBus<Sample> SampleBus { get; set; }
+
+        public SampleService(IRetryBusService retryBusService)
+        {
+            _retryBusService = retryBusService;
+
+            RetryStrategyConfiguration sampleConfiguration = new RetryStrategyConfiguration
+            {
+                Channels = SampleChannels.GetAllChannels(),
+                BusName = BusName,
+                DelaysInMsBetweenEachRetry = new List<int>
+                    {
+                        {5000 },
+                        {6000 },
+                        {10000 }
+                    },
+                MessageIsGoingIntoErrorBusAfterAllRepeat = true,
+                DeliveryDelayInMilliseconds = 1000
+            };
+
+            SampleBus = _retryBusService.CreateNewBus<Sample>(sampleConfiguration);
+        }
+
+        public void Send(Sample sample)
+        {
+            SampleBus.Publish(SampleChannels.MessageSended, sample, true);
+        }
+
+        public void Consume(EventHandler<Sample> callback)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename parameter 'callback' to 'onSampleReceived' to match the interface declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)